tests/unit/aiplatform/test_cloud_training_job.py

```diff
@@ -0,0 +1,224 @@
+
+# CMEK encryption
+_TEST_DEFAULT_ENCRYPTION_KEY_NAME = "key_default"
+_TEST_DEFAULT_ENCRYPTION_SPEC = gca_encryption_spec_compat.EncryptionSpec(
+    kms_key_name=_TEST_DEFAULT_ENCRYPTION_KEY_NAME
+)
+
+_TEST_SERVICE_ACCOUNT = "vinnys@my-project.iam.gserviceaccount.com"
+
+_TEST_NETWORK = f"projects/{_TEST_PROJECT}/global/networks/{_TEST_ID}"
+
+_TEST_TIMEOUT = 8000
+_TEST_RESTART_JOB_ON_WORKER_RESTART = True
+
+_TEST_BASE_CUSTOM_JOB_PROTO = gca_custom_job_compat.CustomJob(
+    display_name=_TEST_DISPLAY_NAME,
+    job_spec=gca_custom_job_compat.CustomJobSpec(
+        worker_pool_specs=_TEST_WORKER_POOL_SPEC,
+        base_output_directory=gca_io_compat.GcsDestination(
+            output_uri_prefix=_TEST_STAGING_BUCKET
+        ),
+        scheduling=gca_custom_job_compat.Scheduling(
+            timeout=duration_pb2.Duration(seconds=_TEST_TIMEOUT),
+            restart_job_on_worker_restart=_TEST_RESTART_JOB_ON_WORKER_RESTART,
+        ),
+        service_account=_TEST_SERVICE_ACCOUNT,
+        network=_TEST_NETWORK,
+    ),
+    encryption_spec=_TEST_DEFAULT_ENCRYPTION_SPEC,
+)
+
+
+@pytest.fixture
+def create_custom_job_mock():
+    with mock.patch.object(
+        job_service_client.JobServiceClient, "create_custom_job"
+    ) as create_custom_job_mock:
+        create_custom_job_mock.return_value = _get_custom_job_proto(
+            name=_TEST_CUSTOM_JOB_NAME,
+            state=gca_job_state_compat.JobState.JOB_STATE_PENDING,
+        )
+        yield create_custom_job_mock
+
+
+@pytest.fixture
+def get_custom_job_mock():
+    with patch.object(
+        job_service_client.JobServiceClient, "get_custom_job"
+    ) as get_custom_job_mock:
+        get_custom_job_mock.side_effect = [
+            _get_custom_job_proto(
+                name=_TEST_CUSTOM_JOB_NAME,
+                state=gca_job_state_compat.JobState.JOB_STATE_PENDING,
+            ),
+            _get_custom_job_proto(
+                name=_TEST_CUSTOM_JOB_NAME,
+                state=gca_job_state_compat.JobState.JOB_STATE_RUNNING,
+            ),
+            _get_custom_job_proto(
+                name=_TEST_CUSTOM_JOB_NAME,
+                state=gca_job_state_compat.JobState.JOB_STATE_SUCCEEDED,
+            ),
+        ]
+        yield get_custom_job_mock
+
+
+def _get_custom_job_proto(state=None, name=None, error=None, version="v1"):
+    custom_job_proto = copy.deepcopy(_TEST_BASE_CUSTOM_JOB_PROTO)
+    custom_job_proto.name = name
+    custom_job_proto.state = state
+    custom_job_proto.error = error
+
+    if version == "v1beta1":
+        v1beta1_custom_job_proto = gca_custom_job_v1beta1.CustomJob()
+        v1beta1_custom_job_proto._pb.MergeFromString(
+            custom_job_proto._pb.SerializeToString()
+        )
+        custom_job_proto = v1beta1_custom_job_proto
+        custom_job_proto.job_spec.tensorboard = _TEST_TENSORBOARD_NAME
+
+    return custom_job_proto
+
+
+class LinearRegression(VertexModel): 
```

Comment:
This mock model should most likely not have a dependency on pytorch. We can add a test case specifically for pytorch models in a follow up PR.

Reply:
What do you recommend I use instead?

Reply:
Let's go with pytorch. Add a `vertex_model_extras_require` to this file: https://github.com/googleapis/python-aiplatform/blob/master/setup.py

And include it in the full installation. It will be installed in the build.

Another option is to add `pytorch` to this line: https://github.com/googleapis/python-aiplatform/blob/master/setup.py#L42